ext/opcache/jit/zend_jit_trace.c

```diff
@@ -6398,6 +6398,9 @@ static const void *zend_jit_trace(zend_jit_trace_rec *trace_buffer, uint32_t par
 			call->prev = frame->call;
 			if (!(p->info & ZEND_JIT_TRACE_FAKE_INIT_CALL)) {
 				TRACE_FRAME_SET_LAST_SEND_BY_VAL(call);
+				if (init_opline && init_opline->opcode == ZEND_INIT_DYNAMIC_CALL) {
```

Comment:
Why is INIT_DYNAMIC_CALL sufficient? Can't you dynamically call a non-closure as well?

Reply:
INIT_DYNAMIC call stay on trace only for closures. Otherwise it exits to VM.